getyour/app/signals.py

```diff
@@ -43,3 +43,46 @@ def household_pre_save(sender, instance, **kwargs):
     if users_household:
         # Save or perform operations with the original instance
         household_history.save()
+
+
+@receiver(pre_save, sender=User)
+def user_pre_save(sender, instance, **kwargs):
+    try:
+        # Save the user's current user data to the database in the
+        # user history table.
+        users_user = User.objects.get(id=instance.id)
```

Comment:
@TiCampbell - This may be a non-issue, but should we save to whole user model to the history table, or only the fields edited by the user?

Reply:
@penoptech I actually will need to reference just the updated fields as part of the extract, so let's please save just the fields edited by the user! Let's do that in all `Hist` tables, if possible.

Reply:
@TiCampbell - I haven't tested this, but we could do a comparison of what is going to be saved to what is currently in the database, find the differences, then store that in the json field. (Code below has no warranty 😂)

```
previous_instance = sender.objects.get(pk=instance.pk)

# Compare the current instance with the previous instance
changed_fields = []

for field in instance._meta.fields:
    field_name = field.name

    if getattr(instance, field_name) != getattr(previous_instance, field_name):
        changed_fields.append(field_name)

household_history = HouseholdHist.objects.create(
    user=instance.user,
    # Convert the household object to a dictionary and then to a JSON string
    # and set it to the historical_household field
    historical_values=json.loads(json.dumps(
        model_to_dict(changed_fields), cls=DjangoJSONEncoder))
)
```

Reply:
Love it! You could have charged more if you had included the warranty :)